src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/controller/v2/FlowControllerV2.java

```diff
@@ -204,6 +211,49 @@
         return flowService.deleteFlowLoop(flowId);
     }
 
+    /**
+     * Gets flow history.
+     */
+    @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+    @ApiOperation(value = "Gets flow status timestamps for flow", responseContainer = "Map")
+    @GetMapping(path = "/{flow_id}/history/statuses")
+    public CompletableFuture<ResponseEntity<SortedMap<Instant, String>>> getFlowStatusTimestamps(
```

Comment:
Inconvenient not expandable response format.

Reply:
Response format has been changed.